tests/integration/operator/ginkgo/suite_test.go

```diff
@@ -0,0 +1,147 @@
+package controllers_test
+
+import (
+	"context"
+	"os"
+
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+
+	//ctrl "sigs.k8s.io/controller-runtime"
+	porterv1 "get.porter.sh/operator/api/v1"
+	//k8s "k8s.io/client-go/kubernetes"
+	//"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var k8sClient client.Client
+
+var testEnv *envtest.Environment
+
+func TestKubernetesPlugin(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		UseExistingCluster: pointer.BoolPtr(true),
+	}
+
+	cfg, err := testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	Expect(scheme.AddToScheme(scheme.Scheme)).To(Succeed())
+	Expect(porterv1.AddToScheme(scheme.Scheme)).To(Succeed())
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+func createTestNamespace(ctx context.Context) string {
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "ginkgo-tests-",
+			Labels: map[string]string{
+				"porter.sh/testdata": "true",
+			},
+		},
+	}
+	Expect(k8sClient.Create(ctx, ns)).To(Succeed())
+
+	// porter-agent service account
+	svc := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "porter-agent",
+			Namespace: ns.Name,
+		},
+	}
+	Expect(k8sClient.Create(ctx, svc)).To(Succeed())
+
+	// Configure rbac for porter-agent
```

Comment:
When we add configmap support to this plugin, I think we'll have to adjust the default porter-agent RBAC to include reading configmaps.

Reply:
Think these were the last ones..

Reply:
@carolynvs This will include the RBAC in the operator installer as well right? Or is that already got enough RBAC to grab config-maps?

Reply:
Right now the agent role has access to secrets but not config maps. So when we add support for config map the operator rbac will need to be updated to grant that as well to the porter-agent role.